### PR TITLE
fluent.syntax.serializer

### PR DESCRIPTION
**Pull request by stasm**
This PR adds the resource serializer to the `fluent.syntax` module.

### CODE REVIEW

File: fluent/syntax/serializer.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        parts.append(serialize_variant(variant, indent + 1))
+
+    parts.append("\n{}".format(" " * 4 * indent))
+
+    return "".join(parts)
+
+
+def serialize_variant(variant, indent=0):
+    return "\n{}{}[{}]{}".format(
+        " " * (4 * indent - 1),
```
Comment (zbraniecki) [inline fluent/syntax/serializer.py]: I'd recommend going for 3 whitespaces here and teh fourth will come from the if_default.

File: tools/prettify.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/python
```
Comment (stasm) [inline tools/prettify.py]: Sure, I like that better, actually.

File: fluent/syntax/serializer.py (hunk 1 above)
Comment (stasm) [inline fluent/syntax/serializer.py]: But that doesn't give the same result when `indent` is greater than 1, right?

Comment (zbraniecki) [inline fluent/syntax/serializer.py]: oh, sorry, I mean, I'd prefer that non-default variants be `"    " * indent`. and default `*` be `"    " * (indent - 1) + "   *"`.

Comment (stasm) [inline fluent/syntax/serializer.py]: I don't understand :( Are you saying that you'd prefer to combine the first `{}{}` in the pattern string into a single one?

Comment (zbraniecki) [inline fluent/syntax/serializer.py]: Haha, ok. So, your code does this:
```
key = {
|||||[one] One
||||*[two] Two
}
```

what I'd like to see is:
```
key = {
||||[one] One
|||*[two] Two
}
```

Does it illustrate it better?

Comment (stasm) [inline fluent/syntax/serializer.py]: Yes, thanks! Except I'm pretty sure it already does the latter and I have tests to prove it :)  It inserts `(4 * indent - 1)` spaces and then adds `*` or an empty space.

Comment (zbraniecki) [inline fluent/syntax/serializer.py]: ugh, f**n operator precedence! Thanks :)

**Pull request merged**